cpp/src/arrow/ipc/read_write_test.cc

```diff
@@ -1519,6 +1519,23 @@ class ReaderWriterMixin : public ExtensionTypesMixin {
     }
   }
 
+  void TestWriteAfterClose() {
+    // Part of GH-35095.
+    std::shared_ptr<RecordBatch> batch_ints;
+    ASSERT_OK(MakeIntRecordBatch(&batch_ints));
+
+    std::shared_ptr<Schema> schema = batch_ints->schema();
+
+    WriterHelper writer_helper;
+    ASSERT_OK(writer_helper.Init(schema, IpcWriteOptions::Defaults()));
+    ASSERT_OK(writer_helper.WriteBatch(batch_ints));
+    ASSERT_OK(writer_helper.Finish());
+
+    // Write after close raises status
+    auto foo = writer_helper.WriteBatch(batch_ints);
+    // ASSERT_RAISES(Invalid, writer_helper.WriteBatch(batch_ints));
```

Comment:
Why is this assertion commented out?

Reply:
Whoops. An extremely embarrassing mistake. Thank you for catching it. It's been fixed.